lib/src/components/NxProgressBar/types.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { HTMLAttributes } from 'react';
+
+export const NX_PROGRESS_BAR_VARIANTS = ['inline', 'small', 'normal', 'full'] as const;
+export type NX_PROGRESS_BAR_VARIANT_TYPE = (typeof NX_PROGRESS_BAR_VARIANTS)[number];
+
+interface BaseProps extends HTMLAttributes<Omit<HTMLProgressElement, 'value' | 'max'>> {
+  inlineCounter?: boolean | null;
+  label: string;
+  labelSuccess?: string | null;
+  max?: number | null;
+  showCounter?: boolean | null;
+  value: number;
+  variant?: NX_PROGRESS_BAR_VARIANT_TYPE | null;
+}
+
+interface PropsWithError extends BaseProps {
+  hasError: true;
+  labelError: string;
+}
+
+interface PropsWithoutError extends BaseProps {
+  hasError?: false | null;
+  labelError?: null;
```

Comment:
Do we need both of these props? If the `labelError` is always defined when there's an error, and also only defined when there's an error, then `hasError` is redundant